sessionctx/variable/session.go

```diff
@@ -1834,6 +1837,44 @@ func (s *SessionVars) GetTemporaryTable(tblInfo *model.TableInfo) tableutil.Temp
 	return nil
 }
 
+// EncodeSessionStates saves session states into SessionStates.
+func (s *SessionVars) EncodeSessionStates(ctx context.Context, sessionStates *sessionstates.SessionStates) (err error) {
+	// Encode user-defined variables.
+	func() {
+		s.UsersLock.RLock()
```

Comment:
As long as there is no early return for this code block, could you remove the `func()` and `defer`?